lib/src/model/user_agent.dart

```diff
@@ -1,19 +1,28 @@
 import '../interface/json_object.dart';
 
 class UserAgent extends JsonObject {
-  final String? name;
+  /// The name of your application (eg: smooth-app)
+  final String name;
+
+  /// The version of the application (1.0.0)
   final String? version;
+
+  /// The system running the application (eg: Android+10)
   final String? system;
+
+  /// The url of your application (eg: https://example.com)
   final String? url;
+
+  /// Additional information about your application
   final String? comment;
 
   const UserAgent({
-    this.name,
+    required this.name,
     this.version,
     this.system,
     this.url,
     this.comment,
-  });
+  }) : assert(name.length > 1, 'A non empty name is required');
```

Comment:
```suggestion
  }) : assert(name.trim().isNotEmpty, 'A non empty name is required');
```

Reply:
Initially that was my change, but calling trim() will require to remove the `const` constructor.
I have no strong opinion on this

Reply:
I have a stronger opinion than you on that matter then: I prefer the `trim` that will reject junk data (and we may make the rules evolve in the future), to the usage of `const` and its highly hypothetical performance gains.

Reply:
Ok, I will `trim` instead